Sources/Testing/Running/EntryPoint.swift

```diff
@@ -10,22 +10,121 @@
 
 private import TestingInternals
 
-/// The entry point to the testing library used by Swift Package Manager.
+#if canImport(Foundation)
+/// The type of the entry point to the testing library used by tools that want
+/// to remain version-agnostic regarding the testing library.
+///
+/// - Parameters:
+///   - argumentsJSON: A buffer to memory representing the JSON encoding of an
+///     instance of `__CommandLineArguments_v0`. If `nil`, a new instance is
+///     created from the command-line arguments to the current process.
+///   - eventHandler: An event handler to which is passed buffers to memory
+///     representing each event and its context, as with ``Event/Handler``, but
+///     encoded as JSON.
 ///
 /// - Returns: The result of invoking the testing library. The type of this
 ///   value is subject to change.
 ///
 /// This function examines the command-line arguments to the current process
 /// and then invokes available tests in the current process.
 ///
+/// - Warning: This function's signature and the structure of its JSON inputs
+///   and outputs have not been finalized yet.
+@_spi(ForToolsIntegrationOnly)
+public typealias ABIEntryPoint_v0 = @Sendable (
+  _ argumentsJSON: UnsafeRawBufferPointer?,
+  _ eventHandler: @escaping @Sendable (_ eventJSON: UnsafeRawBufferPointer, _ contextJSON: UnsafeRawBufferPointer) -> Void
+) async -> CInt
+
+/// Get the entry point to the testing library used by tools that want to remain
+/// version-agnostic regarding the testing library.
+///
+/// - Parameters:
+///   - outEntryPoint: Uninitialized memory large enough to hold an instance of
+///     ``ABIEntryPoint_v0``. On return, a pointer to an instance of that type
+///     representing the ABI-stable entry point to the testing library. The
+///     caller owns this memory and is responsible for deinitializing and
+///     deallocating it when done.
+///
+/// This function can be used by tools that do not link directly to the testing
+/// library and wish to invoke tests in a binary that has been loaded into the
+/// current process. The function is emitted into the binary under the name
+/// `"swt_copyABIEntryPoint_v0"` and can be dynamically looked up at runtime
+/// using `dlsym()` or a platform equivalent.
+///
+/// - Warning: This function's signature and the structure of its JSON inputs
+///   and outputs have not been finalized yet.
+@_cdecl("swt_copyABIEntryPoint_v0")
```

Comment:
This function needs to be surrounded by `#if SWIFT_PACKAGE`

Reply:
We don't support building swift-testing using other build systems like CMake, so I think that would be redundant. We can chat more later today when you get in.